lib/gui/modules/drive-scanner.js

```diff
@@ -110,9 +97,20 @@ const scan = () => {
       includeSystemDrives: settings.get('unsafeMode')
     },
     usbboot: {
-      files: {
-        'bootcode.bin': USBBOOT_BOOTCODE_BIN_BUFFER,
-        'start.elf': USBBOOT_START_ELF_BUFFER
+      readFile: (name) => {
+        if (BLOBS_CACHE[name]) {
+          return Bluebird.resolve(BLOBS_CACHE[name])
+        }
+
+        const blobPath = _.includes([
+          'bootcode.bin',
+          'start_cd.elf',
+          'fixup_cd.dat'
+        ], name) ? path.join('raspberrypi', name) : name
+
+        return fs.readFileAsync(path.join(BLOBS_DIRECTORY, 'usbboot', blobPath)).tap((buffer) => {
+          BLOBS_CACHE[name] = buffer
+        })
```

Comment:
Same as above regarding the device specific config here

Reply:
Also the config being a function feels a bit off; what do you think about moving this into the usbboot adapter?

Reply:
We could move the logic to the adapter, and make the user pass a path from where files will be read as needed, but then it means that we lose the potential flexibility of streaming the files from somewhere else (something we might want to do at some point), or any other way of reading the files.

Reply:
Ah, we'll figure something out for the streams. We could still pass them through the options, as they're one-time consumables anyway, for example.

Reply:
Done, take a look now!